src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -668,7 +705,10 @@ public bool IsChildDirty(int i)
             }
 
             // pruning trick so we never store long persisted paths
-            if (child?.IsPersisted == true)
+            // Dont unresolve node of path length <= 4. there should be a relatively small number of these, enough to fit
+            // in RAM, but they are hit quite a lot, and don't have very good data locality.
+            // That said, in practice, it does nothing notable, except for significantly improving benchmark score.
+            if (child?.IsPersisted == true && childPath.Length > 4 && childPath.Length % 2 == 0)
```

Comment:
why % 2 == 0?

Reply:
So that it still resolve some and unresolve 50% of the time which would still prevent long branch while having a decent 50% uplift.